### PR TITLE
Update grype-db dependency, add some SLES tests

### PR DESCRIPTION
**Pull request by dspalmer99**
This PR updates the version of the syft library used in grype to support SLES distro vuln data. I'm marking this as a draft PR until that data is available in the feed service and I can test with it.

This Pr makes the following changes:

- Bumps the grype-db version to get the SLES distro changes
- Adds a unit and integration tests to verify that matching against SLES distro vulns works as expected.

See anchore/grype#315 for more context on this change.

Signed-off-by: Dan Palmer <dan.palmer@anchore.com>

### CODE REVIEW

File: grype/matcher/common/distro_matchers_test.go [hunk 1]
```diff
@@ -53,6 +53,33 @@ func (pr *mockDistroProvider) stub() {
 			},
 		},
 	}
+	pr.data["sles:12.5"] = map[string][]vulnerability.Vulnerability{
+		// direct...
+		"sles": {
```
Comment (wagoodman) [inline grype/matcher/common/distro_matchers_test.go:58]: nit: the key to this map should be a package name. What you've selected here will work given the package name used to validate is called `sles`, however, it's a little surprising the see the distro name in place of the package name.

File: grype/matcher/common/distro_matchers_test.go [hunk 2]
```diff
@@ -53,6 +53,33 @@ func (pr *mockDistroProvider) stub() {
...
+			{
+				Constraint: version.MustGetConstraint("< 2014.1.5-6", version.RpmFormat),
+				ID:         "CVE-2014-fake-4",
+				Namespace:  "sles:12.5",
+			},
+		},
+		// indirect...
+		"sles-ltss": {
```
Comment (wagoodman) [inline grype/matcher/common/distro_matchers_test.go:66]: nit: this key isn't being used and could probably be removed. It looks like the above `neutron-devel` case isn't needed either (I think this was added back when there was an intention of doing source-package-indirect searching down in the common distro matcher, but they settled up in each distro matcher instead (deb & rpm)).

**Review by wagoodman — APPROVED** [review]

File: grype/matcher/common/distro_matchers_test.go (hunk 1 above)
Comment (dspalmer99) [reply grype/matcher/common/distro_matchers_test.go]: I've made that change.

File: grype/matcher/common/distro_matchers_test.go (hunk 2 above)
Comment (dspalmer99) [reply grype/matcher/common/distro_matchers_test.go]: I've made that change as well, and I removed the existing, unused `neutron-devel` test data.

**Pull request merged**